src/api/libcellml/types.h

```diff
@@ -235,6 +236,86 @@ class LIBCELLML_EXPORT VariablePair
  *  - CellmlElementType::UNITS => std::any_cast<UnitsPtr>.
  *  - CellmlElementType::VARIABLE => std::any_cast<VariablePtr>.
  */
-using AnyItem = std::pair<CellmlElementType, std::any>;
+class LIBCELLML_EXPORT AnyItem
+{
+public:
+    ~AnyItem(); /**< Destructor. */
+    AnyItem(const AnyItem &rhs) = delete; /**< Copy constructor. */
+    AnyItem(AnyItem &&rhs) noexcept = delete; /**< Move constructor. */
+    AnyItem &operator=(AnyItem rhs) = delete; /**< Assignment operator. */
+
+    static AnyItemPtr create() noexcept;
+    /**
+     * @brief Create an AnyItem object.
+     *
+     * Factory method to create a @ref AnyItemPtr.  Create with @c std::any item
+     * and type with::
+     *
+     *   auto anyItem = libcellml::AnyItem::create(item, type);
+     *
+     * @return A smart pointer to a @ref AnyItem object.
+     */
+    static AnyItemPtr create(CellmlElementType type, const std::any &item) noexcept;
+    // KRM
+    static AnyItemPtr create(CellmlElementType type, const ComponentPtr &item) noexcept;
+    static AnyItemPtr create(CellmlElementType type, const ImportSourcePtr &item) noexcept;
+    static AnyItemPtr create(CellmlElementType type, const ModelPtr &item) noexcept;
+    static AnyItemPtr create(CellmlElementType type, const ResetPtr &item) noexcept;
+    static AnyItemPtr create(CellmlElementType type, const UnitPtr &item) noexcept;
+    static AnyItemPtr create(CellmlElementType type, const UnitsPtr &item) noexcept;
+    static AnyItemPtr create(CellmlElementType type, const VariablePtr &item) noexcept;
+    static AnyItemPtr create(CellmlElementType type, const VariablePairPtr &item) noexcept;
```

Comment:
All of these require Doxygen documentation.

Reply:
Not relevant anymore.